go/cli/mcap/utils/checksumming_reader_counter.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"hash"
+	"hash/crc32"
+	"io"
+)
+
+type crcReader struct {
+	r   io.Reader
+	crc hash.Hash32
+}
+
+func (r *crcReader) Read(p []byte) (int, error) {
+	n, err := r.r.Read(p)
+	if n > 0 {
+		_, _ = r.crc.Write(p[:n])
+	}
+	return n, err
+}
+
+func (r *crcReader) Checksum() uint32 {
+	return r.crc.Sum32()
+}
+
+func (r *crcReader) Reset() {
+	r.crc = crc32.NewIEEE()
+}
+
+func newCRCReader(r io.Reader) *crcReader {
+	return &crcReader{
+		r:   r,
+		crc: crc32.NewIEEE(),
+	}
+}
+
+type ChecksummingReaderCounter struct {
+	r     io.Reader
```

Comment:
why do we have two levels of pointer indirection here? for `checksumming_reader_counter.go`, there's just one struct which contains both the count and the hasher directly.